utils/sl/env_modules.py

```diff
@@ -163,12 +164,23 @@ def show_avail(self):
 
 def load_mpi(mpi):
     """global function to load MPI into os.environ"""
-    if _env_module.env_module_init is None:
-        _env_module.env_module_init = _env_module()
-    return _env_module.env_module_init.load_mpi(mpi)
+    # On Ubuntu, MPI stacks use alternatives and need root to change their
+    # pointer, so just verify that the desired MPI is laoded
+    if distro.id() == "ubuntu":
+        try:
+            proc = Popen(['update-alternatives','--query', 'mpi'], stdout=PIPE)
+        except OSError as error:
+            print("Error running update-alternatives")
+            if error.errno == errno.ENOENT:
+                return False
+        for line in proc.stdout.readlines():
+            if line.startswith(b"Value:"):
+                if line[line.rfind(b".")+1:-1].decode() == mpi:
+                    return True
+                else:
+                    return False
+        return False
 
-def list_mpis(mpi):
-    """global function to return list of MPIs"""
     if _env_module.env_module_init is None:
         _env_module.env_module_init = _env_module()
-    return _env_module._mpi_map[mpi]
+    return _env_module.env_module_init.load_mpi(mpi)
```

Comment:
(pylint-missing-final-newline) Final newline missing
(pylint-mixed-line-endings) Mixed line endings LF and CRLF